### PR TITLE
Implement rotation getter and setter for dummy elements

### LINKED ISSUE #2165: Elements created by createElement can't have their rotation changed
**Describe the bug**
Elements created by `createElement` can have their position changed, but rotation isn't working. `setElementRotation` on the element returns `false`, and using OOP syntax `element.rotation = Vector3(...)` doesn't make a difference either, as `element.rotation` always returns `vector3: { x = 0.000, y = 0.000, z = 0.000 } [userdata]`. This also causes custom map elements to not have their rotation applied, even though in EDF they have rotation property set.

**To reproduce**
```lua
element = createElement("dummyType")
v = Vector3(8, 16, 32)
print(setElementPosition(element, v)) -- true
print((element.position - v):getLength() == 0) -- true

print(setElementRotation(element, v)) -- false
print((element.rotation - v):getLength() == 0) -- false
```

**Expected behaviour**
Rotation should be modifiable and saved.

**Screenshots**
not applicable

**Version**
Multi Theft Auto v1.5.8-release-20802

**Additional context**
[Discussion with @Pirulax on Discord](https://discord.com/channels/278474088903606273/667831778563719182/826684876995559434)

### PR DESCRIPTION
**Pull request by ricksterhd123**
When merged this PR will fix #2165

I am still currently working on this PR because OOP is causing my tests to fail. 
Once finished I will post what tests in Lua I've done. I also think that CClientGroups should be refactored to CClientDummyManager in the future to be consistent.

### CODE REVIEW

File: Server/mods/deathmatch/logic/CDummy.cpp [hunk 1]
```diff
@@ -48,12 +48,40 @@ void CDummy::Unlink()
...
 {
     // Grab the position data
     GetCustomDataFloat("posX", m_vecPosition.fX, true);
     GetCustomDataFloat("posY", m_vecPosition.fY, true);
     GetCustomDataFloat("posZ", m_vecPosition.fZ, true);
 
+    // Grab the rotation data
+    GetCustomDataFloat("rotX", m_vecRotation.fX, true);
+    GetCustomDataFloat("rotY", m_vecRotation.fY, true);
+    GetCustomDataFloat("rotZ", m_vecRotation.fZ, true);
```
Comment (Pirulax) [inline Server/mods/deathmatch/logic/CDummy.cpp:85]: Might break something?
Perhaps add a version check maybe. Not sure.

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 2]
```diff
@@ -1204,6 +1215,24 @@ bool CStaticFunctionDefinitions::SetElementPosition(CElement* pElement, const CV
     return true;
 }
 
+bool CStaticFunctionDefinitions::SetDummyRotation(CElement* pElement, const CVector& vecRotation)
```
Comment (Pirulax) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1218]: Is this even used anywhere?

**Comment by Pirulax** [comment]
Have you tested the stuff with older server + current client and vice versa?
It should work.
Other than that it looks fine to me.
Thanks!

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp (hunk 2 above)
Comment (ricksterhd123) [reply Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1218]: Used in SetElementRotation 
https://github.com/multitheftauto/mtasa-blue/blob/0dfac0a08005fd053d8b56fddf0a8ce047d6b859/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L1285

https://github.com/multitheftauto/mtasa-blue/blob/0dfac0a08005fd053d8b56fddf0a8ce047d6b859/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L1290

File: Server/mods/deathmatch/logic/CDummy.cpp (hunk 1 above)
Comment (ricksterhd123) [reply Server/mods/deathmatch/logic/CDummy.cpp:85]: I will check what it does I wasn't sure at all so I just copied to make it look consistent

**Comment by ricksterhd123** [comment]
> Have you tested the stuff with older server + current client and vice versa?

I never thought of doing that I will do a few tests when I'm not busy.

I haven't readied for a review yet because there's some issues with the OOP function for setElementRotation but i wasn't sure if that's me or some existing bug.

Cheers

File: Server/mods/deathmatch/logic/CDummy.cpp (hunk 1 above)
Comment (Pirulax) [reply Server/mods/deathmatch/logic/CDummy.cpp:85]: It's map loading stuff AFAIK. Although, I think map elements probably have their position and rotation set to 0 by default, so it shouldn't be an issue.

**Comment by Pirulax** [comment]
Whats the issue? The value inconsistencies you mentioned over at DC? I think it was you.

**Comment by ricksterhd123** [comment]
> 
> 
> Whats the issue? The value inconsistencies you mentioned over at DC? I think it was you.

Yeah it was me, I'm quite busy so I can't do much this week thanks for following up,
here's what I've been testing (with runcode because it's quick and easy to do):

```lua
start: Resource 'runcode' started
-- server-side
OafishGarlic71 executed command: x = createElement("test")
Command executed!
OafishGarlic71 executed command: setElementRotation(x, 90, 180, 360)
Command results: true [boolean]
OafishGarlic71 executed command: getElementRotation(x)
Command results: 90 [number], 180 [number], 0 [number]
OafishGarlic71 executed command: setElementRotation(x, 90, 180, 259)
Command results: true [boolean]
OafishGarlic71 executed command: getElementRotation(x)
Command results: 90 [number], 180 [number], 259 [number]
OafishGarlic71 executed command: x.rotation
Command results: vector3: { x = 270.000, y = 0.000, z = 79.000 } [userdata]
OafishGarlic71 executed command: x:setRotation(90, 180, 90)
Command results: true [boolean]
OafishGarlic71 executed command: x.rotation
Command results: vector3: { x = 270.000, y = 0.000, z = 270.000 } [userdata]
-- client-side
Executing client-side command: x = createElement("test")
Command executed!
Executing client-side command: setElementRotation(x, 90, 180, 259)
Command results: true [boolean]
Executing client-side command: getElementRotation(x)
Command results: 90 [number], 180 [number], 259 [number]
Executing client-side command: x.rotation
Command results: vector3: { x = 90.000, y = 0.000, z = 79.000 } [userdata]
Executing client-side command: x:setRotation(90, 180, 90)
Command results: true [boolean]
Executing client-side command: x.rotation
Command results: vector3: { x = 90.000, y = 0.000, z = 270.000 } [userdata]
```
The sync is fine but OOP won't play nicely for some reason.

```lua
-- Server-side
OafishGarlic71 executed command: x = createElement("test")
Command executed!
OafishGarlic71 executed command: x:setRotation(90, 180, 90)
Command results: true [boolean]
OafishGarlic71 executed command: x.rotation
Command results: vector3: { x = 270.000, y = 0.000, z = 270.000 } [userdata]
-- Client-side
Executing client-side command: x = getElementsByType("test")
Command executed!
Executing client-side command: getElementRotation(x[1])
Command results: 90 [number], 180 [number], 90 [number]
Executing client-side command: x[1].rotation
Command results: vector3: { x = 90.000, y = 0.000, z = 270.000 } [userdata]
```

File: Server/mods/deathmatch/logic/CDummy.cpp (hunk 1 above)
Comment (ricksterhd123) [reply Server/mods/deathmatch/logic/CDummy.cpp:85]: So, I should remove these 3 lines, or keep them?

**Pull request closed without merging**